llama_stack/providers/adapters/safety/together/safety.py

```diff
@@ -0,0 +1,79 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the terms described in the LICENSE file in
+# the root directory of this source tree.
+import pydantic
+from together import Together
+
+import asyncio
+
+from llama_stack.distribution.request_headers import get_request_provider_data
+from .config import TogetherSafetyConfig
+from llama_stack.apis.safety import *
+import logging
+
+class TogetherHeaderInfo(BaseModel):
+    together_api_key: str
+
+
+class TogetherSafetyImpl(Safety):
+    def __init__(self, config: TogetherSafetyConfig) -> None:
+        self.config = config
+
+    async def initialize(self) -> None:
+        pass
+
+    async def run_shield(
+            self,
+            shield_type: str,
+            messages: List[Message],
+            params: Dict[str, Any] = None,
+    ) -> RunShieldResponse:
+        # support only llama guard shield
+
+        if shield_type != "llama_guard":
+            raise ValueError(f"shield type {shield_type} is not supported")
+
+        provider_data = get_request_provider_data()
+        together_api_key = self.config.api_key
+        # @TODO error out if together_api_key is missing in the header
+        if provider_data is not None:
+            if isinstance(provider_data, TogetherHeaderInfo):
+                together_api_key = provider_data.together_api_key
+
+        # messages can have role assistant or user
+        api_messages = []
+        for message in messages:
+            if type(message) is UserMessage:
+                api_messages.append({'role': message.role, 'content': message.content})
+            else:
+                raise ValueError(f"role {message.role} is not supported")
```

Comment:
do you want to ignore such messages from a conversation or error out? erroring out is going to make this somewhat hard to use and will put the onus of filtering on the user.

Reply:
done